store/iavl/store.go

```diff
@@ -41,20 +42,36 @@ type Store struct {
 // LoadStore returns an IAVL Store as a CommitKVStore. Internally, it will load the
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load, or if called with a positive version on an empty tree.
-func LoadStore(db dbm.DB, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
-	return LoadStoreWithInitialVersion(db, id, lazyLoading, 0)
+func LoadStore(db dbm.DB, logger log.Logger, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
+	return LoadStoreWithInitialVersion(db, logger, id, lazyLoading, 0)
 }
 
 // LoadStoreWithInitialVersion returns an IAVL Store as a CommitKVStore setting its initialVersion
 // to the one given. Internally, it will load the store's version (id) from the
 // provided DB. An error is returned if the version fails to load, or if called with a positive
 // version on an empty tree.
-func LoadStoreWithInitialVersion(db dbm.DB, id types.CommitID, lazyLoading bool, initialVersion uint64) (types.CommitKVStore, error) {
+func LoadStoreWithInitialVersion(db dbm.DB, logger log.Logger, id types.CommitID, lazyLoading bool, initialVersion uint64) (types.CommitKVStore, error) {
 	tree, err := iavl.NewMutableTreeWithOpts(db, defaultIAVLCacheSize, &iavl.Options{InitialVersion: initialVersion})
 	if err != nil {
 		return nil, err
 	}
 
+	if tree.IsUpgradeable() && logger != nil {
+		logger.Info(
+			"module's storage upgrade is in progress while loading",
+			"version", initialVersion,
+			"commit", fmt.Sprintf("%X", id),
+			"is_lazy", lazyLoading,
+		)
+		defer func() {
+			if err != nil {
+				logger.Info("failed upgrading while loading storage", "error", err.Error())
+			} else {
+				logger.Info("storage upgrade is now complete")
+			}
+		}()
+	}
+
 	if lazyLoading {
 		_, err = tree.LazyLoadVersion(id.Version)
```

Comment:
Can we add a logger.Debug after the else case for "Finished loading IAVL tree"

No need to gate that behind tree.IsUpgradeable

Reply:
Done. Not logging anything in case of error because the error would be propagated up the call stack anyways